crates/katana/core/src/service/messaging/ethereum.rs

```diff
@@ -2,62 +2,57 @@ use std::collections::HashMap;
 use std::str::FromStr;
 use std::sync::Arc;
 
+use alloy_network::Ethereum;
+use alloy_primitives::{Address, LogData, U256};
+use alloy_provider::{HttpProvider, Provider};
+use alloy_rpc_types::{BlockNumberOrTag, Filter, FilterBlockOption, FilterSet, Log, Topic};
+use alloy_signer_wallet::LocalWallet;
+use alloy_sol_types::{sol, SolEvent};
 use anyhow::Result;
 use async_trait::async_trait;
-use ethers::prelude::*;
-use ethers::providers::{Http, Provider};
-use ethers::types::{Address, BlockNumber, Log};
-use k256::ecdsa::SigningKey;
 use katana_primitives::chain::ChainId;
 use katana_primitives::receipt::MessageToL1;
 use katana_primitives::transaction::L1HandlerTx;
 use katana_primitives::utils::transaction::compute_l1_message_hash;
 use katana_primitives::FieldElement;
-use tracing::{debug, error, trace, warn};
+use tracing::{debug, trace, warn};
 
 use super::{Error, MessagingConfig, Messenger, MessengerResult, LOG_TARGET};
 
-abigen!(
+sol! {
+    #[sol(rpc, rename_all = "camelcase")]
```

Comment:
Is `snake_case` supported here without any side effect? If yes, we would prefer `snake_case`. :+1:

Reply:
Doesn't seem to work as specified here https://github.com/dojoengine/dojo/pull/1693#discussion_r1536872787